### PR TITLE
save last download dir

### PR DESCRIPTION
**Pull request by tabell**
Cache the last-used download directory to save time when adding multiple torrents in a row

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,8 @@ if (TREMOTESF_BUILD_TESTS)
     enable_testing()
 endif()
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (equeim) [inline CMakeLists.txt:18]: That's needed for some IDEs, right? I think it would be better to split this in separate PR (or at least separate commit).

File: src/tremotesf/ui/screens/addtorrent/addtorrentdialog.cpp [hunk 2]
```diff
@@ -233,7 +236,11 @@ namespace tremotesf
         const auto updateUi = [=] {
             const bool enabled = mRpc->isConnected() && (mFilesModel ? mFilesModel->isLoaded() : true);
             if (enabled) {
-                mDownloadDirectoryWidget->updateComboBox(mRpc->serverSettings()->downloadDirectory());
+                if (Settings::instance()->rememberDownloadDir() && Settings::instance()->lastDownloadDirectory() != "") {
```
Comment (equeim) [inline src/tremotesf/ui/screens/addtorrent/addtorrentdialog.cpp:239]: When QString is compared to string literal, new temporary QString object is created from string literal for the purpose of comparison. While in this case it's likely cheap because QString has empty string optimization, calling isEmpty() would be better solution.

**Review by equeim — CHANGES_REQUESTED** [review]
Sorry for the wait and thanks for your work :)

File: CMakeLists.txt (hunk 1 above)
Comment (tabell) [reply CMakeLists.txt:18]: Yup I use it with vim-lsp for code navigation. I'll remove it from this PR but if you're open to adding it then it makes my life easier :-)

File: src/tremotesf/ui/screens/addtorrent/addtorrentdialog.cpp (hunk 2 above)
Comment (tabell) [reply src/tremotesf/ui/screens/addtorrent/addtorrentdialog.cpp:239]: I'll fix this. Thanks for the review.
